src/components/common/FilterMultiSelect/index.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useEffect, useState } from 'react';
+import { Box, Theme, Typography, useTheme } from '@mui/material';
+import { Button, MultiSelect } from '@terraware/web-components';
+import useDeviceInfo from 'src/utils/useDeviceInfo';
+import strings from 'src/strings';
+import { makeStyles } from '@mui/styles';
+
+interface StyleProps {
+  isMobile?: boolean;
+}
+
+const useStyles = makeStyles((theme: Theme) => ({
+  button: {
+    width: (props: StyleProps) => (props.isMobile ? '100%' : 'auto'),
+  },
+  multiSelectStyle: {
+    height: '100%',
+    width: '100%',
+  },
+}));
+
+type FilterMultiSelectProps<T> = {
+  filterLabel: string;
+  initialSelection: T[];
+  onCancel: (finalSelection: T[]) => void;
+  onConfirm: (finalSelection: T[]) => void;
+  options: T[];
+  renderOption: (item: T) => string;
+};
+
+export default function FilterMultiSelect<T>(props: FilterMultiSelectProps<T>): JSX.Element {
+  const { filterLabel, initialSelection, onCancel, onConfirm, options, renderOption } = props;
+  const { isMobile } = useDeviceInfo();
+  const theme = useTheme();
+  const classes = useStyles({ isMobile });
+
+  const [selection, setSelection] = useState(initialSelection);
+  const [multiSelectOptions, setMultiSelectOptions] = useState<Map<T, string>>(new Map());
+
+  useEffect(() => {
+    const optionsMap = new Map<T, string>(options.map((option) => [option, renderOption(option)]));
+    setMultiSelectOptions(optionsMap);
+  }, [options, renderOption]);
+
+  const onAdd = (item: T) => {
+    setSelection([...selection, item]);
+  };
+
+  const onRemove = (item: T) => {
+    const index = selection.findIndex((x) => item === x);
+    const newSelection = [...selection];
+    newSelection.splice(index, 1);
+    setSelection(newSelection);
+  };
+
+  const clearFilters = () => {
```

Comment:
interesting.. i like this behavior more.
in the current UX, reset applies the cleared-filters in the parent.. it's not a transient function. i think that's a bug.. cc @constanzauanini ?

Reply:
btw, if you hit reset and cancel and reopen the filter box,, do we see the original state?

Reply:
See reset behavior in withdrawal log filters

Reply:
the reset/cancel works in vercel

Reply:
>if you hit reset and cancel and reopen the filter box,, do we see the original state?

yup!

Reply:
Not sure if it is a bug, I think we wanted to work like that. Now that there is a Cancel button, it make more sense to don't clear the parent